lib/strings.nix

```diff
@@ -1051,10 +1052,17 @@ rec {
 
     :::
   */
-  escapeShellArg = arg: "'${replaceStrings ["'"] ["'\\''"] (toString arg)}'";
+  escapeShellArg = arg:
+    let
+      string = toString arg;
+    in
+      if match "[[:alnum:],._+:@%/-]*" string == null
```

Comment:
I'm on my phone right now so I can't check, but how does `*` here work? Would is not also match zero instances of those chars?

Reply:
Good catch. We need `+` here (and a test case that exercises the empty string). That probably would have broken something…